### ISSUE TITLE
Add type hints to better understand source

### ISSUE THREAD

**Issue by NicoHood** [niccokunzmann/python-recurring-ical-events#91]
I currently try to find out if the `between()` function can take a `date` and/or `datetime` object. I think it can parse both (as that's what I remember I've tested), but I am not sure. Also I do not understand the source with that tuple and int instance checking, thats confusing to me.

Could you please clarify what the accepted inputs of `between()` are and maybe add this as a typehint?

---
<small>
We're using [Polar.sh](https://polar.sh/niccokunzmann) so you can upvote and help fund this issue. We receive the funding once the issue is completed & confirmed by you. Thank you in advance for helping prioritize & fund our work.
</small>

<a href="https://polar.sh/niccokunzmann/python-recurring-ical-events/issues/91">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/niccokunzmann/python-recurring-ical-events/issues/91/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/niccokunzmann/python-recurring-ical-events/issues/91/pledge.svg">
</picture>
</a>

**Comment by niccokunzmann** [comment]
I think, the README file describes which kinds of data structures you can pass to the between function.
I did not use type-hints, yet.

The motivation behind this is to be able to pass hard-coded arguments fast without needing to put them though a datetime.date or datetime.datetime on the user side.

**Comment by NicoHood** [comment]
Ah, now I see. I would not do that, but as long as date and datetime works, thats fine for me.

**Comment by niccokunzmann** [comment]
yes, the end date is excluded. That should be documented this way if I remember correctly.

**Comment by niccokunzmann** [comment]
It seems that combine is not present. 

```
Python 3.9.5 (default, Nov 23 2021, 15:27:38)
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> help("datetime.combine")
No Python documentation found for 'datetime.combine'.
```

**Comment by niccokunzmann** [comment]
Thanks! I could have thought about that :,)
How would you formulate this issue so that it has closing criteria? Is it "Added type hints to at(), between() and of()"?

**Comment by niccokunzmann** [comment]
Hm. I just saw this example concerning py.typed: https://github-redirect.dependabot.com/certifi/python-certifi/issues/196 That reminded me of this issue.

**Comment by niccokunzmann** [comment]
If someone likes to add type hints, this is welcome. I will support that but not do it myself at the moment.

**Comment by NicoHood** [comment]
Why? It is so much easier to understand your own code. At least for newer code this should be done. You can also then do typechecks using pylance in vscode.

**Comment by niccokunzmann** [comment]
I do it in my free time, I do not get paid and I do not see the value. Other people see value, have time. I am more in survival mode at the moment. When a basic income is secured I can gift my time. Yes, it might take me less time and it would be efficient if I did it and I would learn but under the circumstances, I choose to work on other issues.

**Comment by niccokunzmann** [comment]
This is fixed in v3.0.0
